### PR TITLE
StorageFolder.CreateFileAsync(...) fixes

### LINKED ISSUE #1067: CreationCollisionOption.ReplaceExisting is treated as CreationCollisionOption.OpenIfExists
```
 //The following code shows CreationCollisionOption.ReplaceExisting is treated as CreationCollisionOption.OpenIfExists
            StorageFile sfFoo = await ApplicationData.Current.LocalFolder.CreateFileAsync("foo", CreationCollisionOption.ReplaceExisting);
            using (StreamWriter dr = new StreamWriter(await sfFoo.OpenStreamForWriteAsync()))
            {
                await dr.WriteAsync("this is foo + foo");
                await dr.FlushAsync();
                dr.Close(); //Supposed to be redundant
            }
            StorageFile sfFoo2 = await ApplicationData.Current.LocalFolder.CreateFileAsync("foo", CreationCollisionOption.ReplaceExisting);
            using (StreamWriter dr = new StreamWriter(await sfFoo.OpenStreamForWriteAsync()))
            {
                await dr.WriteAsync("this is abc");
                await dr.FlushAsync();
                dr.Close(); //Supposed to be redundant
            }

            StorageFile sfFooRead = await ApplicationData.Current.LocalFolder.GetFileAsync("foo");
            using (FileStream stream = File.OpenRead(sfFooRead.Path))
            using (StreamReader sr = new StreamReader(stream))
            {
                {
                    //The following outputs: "this is abc + foo"
                    //It is supposed to be: "this is abc"
                    Console.WriteLine(sr.ReadToEnd());
                }
            }

```
**Environment**

- OS:       iOS
- Version   12
- Device:   Simulator

Repro:
https://github.com/zipswich/UnoDebug

### LINKED ISSUE #3123: StorageFolder.CreateFile with CreationCollisionOption.ReplaceExisting does not empty an existing file
## Current behavior

` var storageFile = await ApplicationData.Current.LocalFolder.CreateFileAsync(LocalSettingsKey, CreationCollisionOption.ReplaceExisting);`

will not empty an existing file if it already exists.

## Expected behavior

The file is emptied properly, and new data can be written to it.

## Environment

Nuget Package: 2.3-dev

### PR DESCRIPTION
**Pull request by ebariche**
Fixes #1067, Fixes #3123

## PR Type

What kind of change does this PR introduce?
- Bugfix
- Refactoring (no functional changes, no api changes)

## What is the current behavior?

- StorageFolder.CreateFileAsync(String) throws ArgumentOutOfRangeException if file already exists.
- StorageFolder.CreateFileAsync(String, CreationCollisionOption.ReplaceExisting) behaves like OpenIfExists

## What is the new behavior?

- StorageFolder.CreateFileAsync(String) throws Exception if file already exists. (As per UWP)
- StorageFolder.CreateFileAsync(String, CreationCollisionOption.ReplaceExisting) behaves correctly

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UWP/Storage/StorageFolder.cs [hunk 1]
```diff
@@ -176,42 +176,45 @@ internal StorageFolder(string name, string path)
 
 		public IAsyncOperation<StorageFile> CreateFileAsync(string desiredName) => CreateFileAsync(desiredName, CreationCollisionOption.FailIfExists);
 
-        public IAsyncOperation<StorageFile> CreateFileAsync(string path, CreationCollisionOption option) =>
+        public IAsyncOperation<StorageFile> CreateFileAsync(string desiredName, CreationCollisionOption option) =>
```
Comment (jeromelaban) [inline src/Uno.UWP/Storage/StorageFolder.cs:179]: This is a breaking change. Is it how the parameter is named under UWP ?

File: src/Uno.UWP/Storage/StorageFolder.cs [hunk 2]
```diff
@@ -176,42 +176,45 @@ internal StorageFolder(string name, string path)
...
             AsyncOperation.FromTask(async ct =>
             {
-                if (File.Exists(global::System.IO.Path.Combine(Path, path)))
+                if (File.Exists(global::System.IO.Path.Combine(Path, desiredName)))
                 {
                     switch (option)
                     {
+						case CreationCollisionOption.FailIfExists:
+							throw new Exception("Cannot create a file when that file already exists.");
```
Comment (jeromelaban) [inline src/Uno.UWP/Storage/StorageFolder.cs:187]: I agree with the robot, but it does not matter much if the exception type is not the same as UWP here. We can use a more acceptable exception type, like `InvalidOperation` for instance.

Comment (pkar70) [reply src/Uno.UWP/Storage/StorageFolder.cs:187]: But if someone creates code with catch, and implemented logic for handling this exception, then changing exception that is thrown (another in UWP, another in Uno) breaks such code. I think Uno should be as far UWP as even "bug compatible" - just to not break user code that expects some behaviour, even expects known UWP errors.

File: src/Uno.UWP/Storage/StorageFolder.cs (hunk 1 above)
Comment (ebariche) [reply src/Uno.UWP/Storage/StorageFolder.cs:179]: desiredName

**Pull request merged**